command/run.go

```diff
@@ -107,12 +112,24 @@ func (c *RunCommand) Run(args []string) int {
 		c.Ui.Error(c.Help())
 		return 1
 	}
-	file := args[0]
 
-	// Parse the job file
-	job, err := jobspec.ParseFile(file)
+	// Read the Jobfile
+	path := args[0]
+	var f io.Reader = os.Stdin
+	if path != "-" {
+		file, err := os.Open(path)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error opening file: %s", err))
+			return 1
+		}
+		defer file.Close()
+		f = file
+	}
+
+	// Parse the JobFile
+	job, err := jobspec.Parse(f)
 	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error parsing job file %s: %s", file, err))
+		c.Ui.Error(fmt.Sprintf("Error parsing job file %s: %s", f, err))
```

Comment:
%s of the io.Reader will not be nice output. If it is stdin I would say "parsing job file from stdin" and if a  file "parsing job file from %q"